### PR TITLE
Update libss and libcrypto to universal libraries

### PR DESCRIPTION
**Pull request by Kaspik**
Replacement of https://github.com/Sequel-Ace/Sequel-Ace/pull/412 with only required changes

### CODE REVIEW

**Comment by Kaspik** [comment]
@jamesstout I believe the libcrypto wasn't compiled properly somehow... buidl works, but unit tests don't, and the error is:
`ld: file not found: /tmp/openssl-arm/lib/libcrypto.1.1.dylib for architecture x86_64` - and as you can see in this PR, the only thing that happened was that we replaced the libraries with new ones, nothing else... 😞

**Comment by jamesstout** [comment]
Hmmm .. will take a look

**Comment by jamesstout** [comment]
Well ... it works on Big Sur with Xcode 12. Will have a look at fixing the build

**Comment by Kaspik** [comment]
@jamesstout Mergr main please, there was a change in XCScheme :)

File: build.sh [hunk 1]
```diff
@@ -22,9 +22,34 @@ set -e
...
+then
+  echo "curl of $OPENSSL_URL failed"
+  trap - EXIT
+  exit 1
+fi
+
 if [ "$MODE" = "tests" ]; then
   echo "Running Sequel Ace Unit tests"
-  set -o pipefail && xcodebuild test -project sequel-ace.xcodeproj -scheme "Sequel Ace Local Testing" -destination "platform=macOS,arch=x86_64" test CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO | xcpretty -c
+  set -o pipefail && xcodebuild test -verbose -project sequel-ace.xcodeproj -scheme "Sequel Ace Local Testing" -destination "platform=macOS,arch=x86_64" test CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO | xcpretty -c
```
Comment (Kaspik) [inline build.sh:52]: can we disable the verbose mode now? :)

**Review by Kaspik — COMMENTED** [review]
Looks god to me, haven't tested yet tho!

**Comment by Kaspik** [comment]
@jamesstout 

Screenshot 2020-11-13 at 15 09 31

Removed derived data, tried first build - not successful. Do I have wrong target, or what's up?  They are somehow messed up.
Screenshot 2020-11-13 at 15 09 31

**Comment by Kaspik** [comment]
@Jason-Morcos You changed targets, what's this? 😂

**Comment by jamesstout** [comment]
> Removed derived data, tried first build - not successful. Do I have wrong target, or what's up? They are somehow messed up.

You need to download openssl tar.gz and put it in Frameworks/openssl. Could do that automatically I guess...

`curl -L https://www.openssl.org/source/openssl-1.1.1h.tar.gz -o Frameworks/openssl/openssl-1.1.1h.tar.gz`

**Comment by Kaspik** [comment]
Can we commit it to repo so we have stable version included in the project so people that want to contribute don't have to do it? Or ideally - can we separate this into another project not part of the main one at all, and when there is new OPEN SSL version just download it, run it again, and it will make changes to the main project?

**Comment by Kaspik** [comment]
Okay, made some changes:

- added the `tar.gz` file next to the openssl.xcodeproj so we have it there and we know what version we compiled and it's available for everybody and not part of the CI process to download it
- removed openssl out of main project and left it as a separate project only to rebuild opensll next time we need to upgrade
- disabled the code signing part of the build process as it generates new dylib files every run which is something we don't want... we want to have one and the same dylib file as universal library until we upgrade openssl again

@jamesstout Let me know if you think that the Run script step in SPMySQLFramework is unnecessary and we are good to comment it out even more. But right now:
- to generate new dylib files, you just replace the `tar.gz` file with new version, open opensll.xcodeproj and run it
- to code sign it then you go to main project - SPMySQLFramework build step and un-comment the part to enable code signing to code sign it then, then comment it out again

**Comment by jamesstout** [comment]
> Okay, made some changes:
> 
> * added the `tar.gz` file next to the openssl.xcodeproj so we have it there and we know what version we compiled and it's available for everybody and not part of the CI process to download it
> * removed openssl out of main project and left it as a separate project only to rebuild opensll next time we need to upgrade
> * disabled the code signing part of the build process as it generates new dylib files every run which is something we don't want... we want to have one and the same dylib file as universal library until we upgrade openssl again
> 

Nice one.

> @jamesstout Let me know if you think that the Run script step in SPMySQLFramework is unnecessary and we are good to comment it out even more. But right now:
> 
> * to generate new dylib files, you just replace the `tar.gz` file with new version, open opensll.xcodeproj and run it
> * to code sign it then you go to main project - SPMySQLFramework build step and un-comment the part to enable code signing to code sign it then, then comment it out again

I think we still need the run script to re-write the @loader_paths for the dylibs, but only if they are new...

**Comment by Kaspik** [comment]
I think that's what I left there enabled, so this looks good?

**Comment by jamesstout** [comment]
> I think that's what I left there enabled, so this looks good?

Just got a couple of tweaks I'm testing for the "but only if they are new..." part

**Comment by Kaspik** [comment]
@Jason-Morcos @jamesstout  I believe this is ready for review, I can't do review on my own PR 😄

**Review by Jason-Morcos — APPROVED** [review]
Looks good to me, I think we can merge this to the 3.0 branch and include in 3.0 release.

I might revert the scheme change, however. If we want to revert the previous schemes change, should make it a separate PR. I made the three Schemas on purpose with different prefs for local, beta, and release builds. We can chat about schemas more on slack in the am, but I would say for now might be good to drop the change from this PR and make it a separate one if desired to discuss.

And I do think this should be a 3.0 change! :)

**Comment by Jason-Morcos** [comment]
When I say schema change, I mean the file adding back the "Sequel Ace" target schema that was renamed in another PR to Sequel Ace Local Testing.

**Comment by Kaspik** [comment]
I would merge this to 3.0 and do the schemas there too, regardless of 2.x.x :)

**Comment by Jason-Morcos** [comment]
@Kaspik Switched base branch to 3.0. There's some conflicts to resolve and then we can merge to 3.0. I do still think though we should remove the Sequel Ace.xcscheme files from this PR as, if I understand correctly, it's completely unrelated to the PR.

**Pull request merged**